core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -2420,6 +2420,44 @@ private[spark] object Utils extends Logging {
   }
 }
 
+private[spark] class CallerContext(
+   appName: Option[String] = None,
+   appID: Option[String] = None,
+   appAttemptID: Option[String] = None,
+   jobID: Option[Int] = None,
+   stageID: Option[Int] = None,
+   stageAttemptId: Option[Int] = None,
+   taskId: Option[Long] = None,
+   taskAttemptNumber: Option[Int] = None) extends Logging {
+
+   val AppName = if (appName.isDefined) s"_AppName_${appName.get}" else ""
```

Comment:
thinking about this a little more I think we should just remove app name altogether. For tracking purposes the app id should be enough. The app name adds in a bunch of unknowns (length or it, weird characters or just spaces) which could cause things in the audit log to be less useful.